src/create-client.ts

```diff
@@ -258,11 +261,31 @@ An authorization_code was supplied for a login which did not originate at the ap
   }
 
   async #refreshSession({ organizationId }: { organizationId?: string } = {}) {
+    if (this.#state.tag === "AUTHENTICATING") {
+      await this.#state.response;
+      return;
+    }
+
     const beginningState = this.#state;
-    this.#state = { tag: "AUTHENTICATING" };
 
+    this.#state = {
+      tag: "AUTHENTICATING",
+      response: this.#doRefresh({ organizationId, beginningState }),
+    };
+
+    await this.#state.response;
+    return;
```

Comment:
Nit: No `return` needed since this is already the last statement.

```suggestion
    await this.#state.response;
```